pgmpy/factors/CPD.py

```diff
@@ -368,6 +369,63 @@ def to_factor(self):
         """
         return Factor(self.variables, self.cardinality, self.values)
 
+    def reorder_parents(self, new_order, inplace=False):
+        """
+        Reorders the parents according to the new given order
+
+        Parameters
+        ----------
+        new_order: list
+            list of new ordering of variables
+        inplace: boolean
+            If inplace=True it will modify the CPD itself and return it,
+            otherwise new value will be returned without affecting old values
+
+        Returns
+        -------
+        Returns the new reordered CPD
+
+        Examples
+        --------
+        >>>from pgmpy.factors.CPD import TabularCPD
+        >>> cpd = TabularCPD('J', 2, [[0.9,0.3,0.9,0.3,0.8,0.8,0.4,0.4],
+        ...                          [0.1,0.7,0.1,0.7,0.2,0.2,0.6,0.6]],
+        ...                  evidence=['C', 'B', 'A'], evidence_card= [2, 2, 2])
+        >>> cpd.get_cpd()
+        array([[ 0.9,  0.3,  0.9,  0.3,  0.8,  0.8,  0.4,  0.4],
+                [ 0.1,  0.7,  0.1,  0.7,  0.2,  0.2,  0.6,  0.6]])
+        >>> cpd.reorder_parents(['B', 'A', 'C'])
+        array([[ 0.9,  0.9,  0.3,  0.3,  0.8,  0.4,  0.8,  0.4],
+               [ 0.1,  0.1,  0.7,  0.7,  0.2,  0.6,  0.2,  0.6]])
+        """
+        if not self.evidence or (set(new_order) - set(self.evidence)) or (set(self.evidence) - set(new_order)):
+            raise ValueError("New order either has missing or extra arguments")
+        else:
```

Comment:
Without having that else code would have worked just fine. You are adding an another level of indentation.